### ISSUE TITLE
[BUG][Warning] In normal usage, there are some warnings when using AutoSklearn.

### ISSUE THREAD

**Issue by tsungjung411** [mlflow/mlflow#9109]
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Willingness to contribute

No. I cannot contribute a bug fix at this time.

### MLflow version

- MLflow version: 2.3.2

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 20.04.6 LTS
- **Python version**: Python 3.9.17

### Describe the problem

[Warning] In normal usage, there are some warnings when using AutoSklearn.

>2023/07/18 11:53:02 INFO mlflow.tracking.fluent: Autologging successfully enabled for sklearn.
2023/07/18 11:53:10 INFO mlflow.utils.autologging_utils: Created MLflow autologging run with ID '6ba7b3a5071b49559e68cec5b0bd1ea1', which will track hyperparameters, performance metrics, model artifacts, and lineage information for the current sklearn workflow
2023/07/18 11:53:10 WARNING mlflow.sklearn: Training metrics will not be recorded because training labels were not specified. To automatically record training metrics, provide training labels as inputs to the model training function.
2023/07/18 11:53:10 WARNING mlflow.sklearn: Failed to infer model signature: 'StandardScalerComponent' object has no attribute 'predict'
2023/07/18 11:53:13 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/usr/local/miniconda3/envs/autosklearn/lib/python3.9/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils."
2023/07/18 11:53:14 INFO mlflow.utils.autologging_utils: Created MLflow autologging run with ID 'a0b5b24aef20494e99115710800d785a', which will track hyperparameters, performance metrics, model artifacts, and lineage information for the current sklearn workflow
2023/07/18 11:53:14 WARNING mlflow.sklearn: Training metrics will not be recorded because training labels were not specified. To automatically record training metrics, provide training labels as inputs to the model training function.
2023/07/18 11:53:14 WARNING mlflow.sklearn: Failed to infer model signature: the trained model does not specify a `predict` function, which is required in order to infer the signature
2023/07/18 11:53:14 WARNING mlflow.sklearn: Model was missing function: predict. Not logging python_function flavor!
2023/07/18 11:53:17 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/usr/local/miniconda3/envs/autosklearn/lib/python3.9/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils."

such as:
- WARNING mlflow.sklearn: Failed to infer model signature: 'StandardScalerComponent' object has no attribute 'predict'
- WARNING mlflow.sklearn: Failed to infer model signature: the trained model does not specify a `predict` function, which is required in order to infer the signature

### Tracking information

```shell
$ mlflow server -h 0.0.0.0 -p 5000 --backend-store-uri sqlite:///mlruns.db --default-artifact-root ./mlruns
```

```python
print("MLflow version:", mlflow.__version__)
print("Tracking URI:", mlflow.get_tracking_uri())
print("Artifact URI:", mlflow.get_artifact_uri())
```
```
MLflow version: 2.3.2
Tracking URI: sqlite:///mlruns.db
Artifact URI: /workspace/tmp/mlruns/0/44d64ecde9774519941e58d9c424784f/artifacts
```

### prerequisite
- `from sklearn.ensemble import RandomForestClassifier` requires to install the following package:
   ```
   $ pip install scikit-learn==1.1.2
   ```
- `from autosklearn.classification import AutoSklearnClassifier` requires to install the following package:
   ```
   $ pip install auto-sklearn
   ```

### sample code (demo code)
```python
import mlflow
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score
from autosklearn.classification import AutoSklearnClassifier

mlflow.set_tracking_uri('sqlite:///mlruns.db')
mlflow.autolog()

X, y = load_iris(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y)

option = 2
if option == 1:
    model = RandomForestClassifier()
elif option == 2:
    params = {}
    params['memory_limit'] = 1024 * 12 # (in MB) (default: 3072 MB)
    params['time_left_for_this_task'] = 120
    params['per_run_time_limit'] = 30
    model = AutoSklearnClassifier(**params)

model.fit(X_train, y_train)
y_pred = model.predict(X_test)

# inference
print("score:", model.score(X_test, y_test))
print('acc:', accuracy_score(y_test, y_pred))


# save the model to the local disk
import cloudpickle

with open("model.pkl", "wb") as f:
    cloudpickle.dump(model, f)
    print("the model saved successfully")


# load a model from the local disk
with open("model.pkl", "rb") as f:
    model2 = cloudpickle.load(f)
    print('the model loaded successfully')
    y_pred2 = model2.predict(X_test)
    print('acc:', accuracy_score(y_test, y_pred2))
```

### Stack trace

[Warning] In normal usage, there are some warnings when using AutoSklearn.

>2023/07/18 11:53:02 INFO mlflow.tracking.fluent: Autologging successfully enabled for sklearn.
2023/07/18 11:53:10 INFO mlflow.utils.autologging_utils: Created MLflow autologging run with ID '6ba7b3a5071b49559e68cec5b0bd1ea1', which will track hyperparameters, performance metrics, model artifacts, and lineage information for the current sklearn workflow
2023/07/18 11:53:10 WARNING mlflow.sklearn: Training metrics will not be recorded because training labels were not specified. To automatically record training metrics, provide training labels as inputs to the model training function.
2023/07/18 11:53:10 WARNING mlflow.sklearn: Failed to infer model signature: 'StandardScalerComponent' object has no attribute 'predict'
2023/07/18 11:53:13 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/usr/local/miniconda3/envs/autosklearn/lib/python3.9/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils."
2023/07/18 11:53:14 INFO mlflow.utils.autologging_utils: Created MLflow autologging run with ID 'a0b5b24aef20494e99115710800d785a', which will track hyperparameters, performance metrics, model artifacts, and lineage information for the current sklearn workflow
2023/07/18 11:53:14 WARNING mlflow.sklearn: Training metrics will not be recorded because training labels were not specified. To automatically record training metrics, provide training labels as inputs to the model training function.
2023/07/18 11:53:14 WARNING mlflow.sklearn: Failed to infer model signature: the trained model does not specify a `predict` function, which is required in order to infer the signature
2023/07/18 11:53:14 WARNING mlflow.sklearn: Model was missing function: predict. Not logging python_function flavor!
2023/07/18 11:53:17 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/usr/local/miniconda3/envs/autosklearn/lib/python3.9/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils."

such as:
- WARNING mlflow.sklearn: Failed to infer model signature: 'StandardScalerComponent' object has no attribute 'predict'
- WARNING mlflow.sklearn: Failed to infer model signature: the trained model does not specify a `predict` function, which is required in order to infer the signature
- WARNING mlflow.sklearn: Training metrics will not be recorded because training labels were not specified. To automatically record training metrics, provide training labels as inputs to the model training function.

![image](https://github.com/mlflow/mlflow/assets/14820019/381cbaff-201d-43c3-bb2e-24ac4069f430)

I believe there might be some code that hasn't been handled properly, causing some warnings even in normal usage of AutoSklearn. I'm not sure about the extent of the impact caused by these warnings.

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by harupy** [comment]
@tsungjung411 This is a change in MLflow 2.5.0. In 2.5, MLflow attempts to infer model signatures.

**Comment by harupy** [comment]
Looks like you're using mlflow 2.3.2

**Comment by serena-ruan** [comment]
This does look like a bug: https://github.com/mlflow/mlflow/blob/f2fadddcbb653d4ec7f2ec3367b2f237d56c9f72/mlflow/sklearn/__init__.py#L1309 
'StandardScalerComponent' only has 'transform' function, we should fix it @harupy ?

**Comment by harupy** [comment]
Fixed by #9123
